src/js/lesson_1/homework.js

```diff
@@ -29,8 +44,23 @@ function factorial(number) {
  *
  * @return {string} entered name
  */
+
+let isConf;
+let string;
 function greetAndConfirm() {
-  // ...
+  string = prompt('Enter your name:');
+
+  if (!string) {
+    throw new Error('Provided empty name');
+  } else if (string !== null) {
```

Comment:
Можно заменить вот на такую проверку, будет эквивалентно предложенной вами:
```suggestion
  } else if (!string) {
```